### PR TITLE
Feature/vfs redesign

### PR DESCRIPTION
**Pull request by iamgweej**
NOTE: Don't merge this yet, for some reason I'm not able to run tests right now `zig build test` fails, I suspect it has something to do with https://github.com/ziglang/zig/issues/6493 or something of that sort, working with `zig version 0.6.0+a31b70c4b`)

This should solve https://github.com/ZystemOS/pluto/issues/252.

### CODE REVIEW

File: src/kernel/filesystem/vfs.zig [hunk 1]
```diff
@@ -74,16 +74,16 @@ pub const FileSystem = struct {
     /// Arguments:
     ///     IN self: *const FileSystem - The filesystem in question being operated on
     ///     IN node: *const FileNode - The file being read from
-    ///     IN len: usize - The number of bytes to read from the file
+    ///     IN bytes: []u8 - The buffer to fill data from the file with
     ///
-    /// Return: []u8
-    ///     The data read as a slice of bytes. The length will be <= len, including 0 if there was no data to read
+    /// Return: usize
+    ///     The length of the actuall data read. This being < bytes.len is not considered an error.
```
Comment (SamTebbs33) [inline src/kernel/filesystem/vfs.zig]: We should also mention that it's impossible for the returned value to be greater than the len. Same for the Write type.

File: src/kernel/filesystem/vfs.zig [hunk 2]
```diff
@@ -93,10 +93,13 @@ pub const FileSystem = struct {
     ///     IN node: *const FileNode - The file being read from
     ///     IN bytes: []u8 - The bytes to write to the file
     ///
+    /// Return: usize
+    ///     The length of te actual data written to the file. This being < bytes.len is not considered an error.
```
Comment (SamTebbs33) [inline src/kernel/filesystem/vfs.zig]: te -> the

**Review by SamTebbs33 — CHANGES_REQUESTED** [review]
Looking good!

File: src/kernel/filesystem/vfs.zig (hunk 1 above)
Comment (iamgweej) [reply src/kernel/filesystem/vfs.zig]: Added

File: src/kernel/filesystem/vfs.zig (hunk 2 above)
Comment (iamgweej) [reply src/kernel/filesystem/vfs.zig]: Changed

**Review by SamTebbs33 — APPROVED** [review]
Nice one.

**Comment by SamTebbs33** [comment]
I'll merge if you don't mind squashing them into one commit.

**Comment by SamTebbs33** [comment]
I'll go ahead and squash and merge since I'm going to be doing some work on the vfs and don't want any conflicts with this PR. Thanks again!

**Comment by SamTebbs33** [comment]
I'll squash and merge this as I'm going to be doing some work on the VFS and don't want to get any conflicts. Thanks again!

**Pull request merged**